project/src/com/google/daggerqueryui/scripts/binding_graph.js

```diff
@@ -118,9 +123,29 @@ const bindingGraph = (function() {
     const sourceId = addNode(source);
     const targetId = addNode(target);
 
+    // Checks if an edge already exists.
+    if (isEdgeExists(sourceId, targetId)) {
+      return;
+    }
+
     edges.add({from: sourceId, to: targetId});
   }
 
+  /**
+   * Checks if an edge exists in the graph or not.
+   *
+   * @param {number} sourceId
+   * @param {number} targetId
+   * @return {boolean}
+   */
+  function isEdgeExists(sourceId, targetId) {
```

Comment:
nit: can we rename this to edgeExists? or hasEdge?

Reply:
Oops, I see what's wrong, thank you!  
[Done](https://github.com/googleinterns/dagger-query/pull/45/commits/05f2f30c1e2e9acce8a810fb885325c2b22c0360) 💎